### PR TITLE
Extract MRR data; generate MRR document and attach to package

### PR DESCRIPTION
**Pull request by jasonburkert-ccpo**
- Adds _system property_ `x_g_dis_atat.docgen.mrr.enable`, currently set to `true`
- Adds (and publishes) _action_ `DOCGEN: Gather MRR Data`
- Adds _script include_ `MarketResearchReport`
- Adds (and publishes) _subflow_ `DOCGEN: Generate MRR`

Tickets: AT-8836, AT-8252

### CODE REVIEW

**Comment by jasonburkert-ccpo** [comment]
### Action execution
image

**Comment by jasonburkert-ccpo** [comment]
## Subflow execution
image

## MRR attached to package
image

File: f600233d1b154d507b782f84604bcb12/update/sys_script_include_da81226197072110cf3cfd9fe153afee.xml [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+        // Look up Fair Opportunity record
+        this.fairOpportunity.get(this.acquisitionPackage.fair_opportunity);
+        if (!this.fairOpportunity.sys_id) {
+            throw this.constructError(this.DATA_INCOMPLETE, "No Fair Opportunity record found");
+        }
+
+    },
+
+    explicitNull: function(value) {
```
Comment (jeffsegal-ccpo) [inline f600233d1b154d507b782f84604bcb12/update/sys_script_include_da81226197072110cf3cfd9fe153afee.xml]: Should be able to re-use the J&A version of this (see `this.explicitNull = this.janda.explicitNull;` from `initialize`).

File: f600233d1b154d507b782f84604bcb12/update/sys_script_include_da81226197072110cf3cfd9fe153afee.xml [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+    initialize: function(acqPackage) {
+        this.acquisitionPackage = acqPackage;
+        this.janda = new JustificationAndApproval(acqPackage);
+        this.constructError = this.janda.constructError;
+        this.explicitNull = this.janda.explicitNull;
+
+        this.requirementsChecklist = new RequirementsChecklist(acqPackage);
+
+        // Look up Fair Opportunity record
```
Comment (jeffsegal-ccpo) [inline f600233d1b154d507b782f84604bcb12/update/sys_script_include_da81226197072110cf3cfd9fe153afee.xml]: Should be able to replace this with 

```
this.fairOpportunity = this.janda.fairOpportunity;
```

File: f600233d1b154d507b782f84604bcb12/update/sys_script_include_da81226197072110cf3cfd9fe153afee.xml [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+        try {
+            return getListRecords(techniqueSysIds, PROPERTY_NAMES, REFERENCE_TABLE)
+                .filter(Dow.prototype.atatUtil.recordNotNull)
+                .sort((a, b) => { // OTHER must present last; has sequence 99
+                    return a.sequence - b.sequence;
+                })
+                .map(removeSysId);
+
+        } catch (error) {
+            throw this.errUtil.createError(
```
Comment (jeffsegal-ccpo) [inline f600233d1b154d507b782f84604bcb12/update/sys_script_include_da81226197072110cf3cfd9fe153afee.xml]: I am open to being proven wrong, but I have not seen any value to the catch/re-throw pattern that prevails in the code base, unless we are actually doing something useful in the throw like translating internal errors to HTTP response codes.

**Review by jeffsegal-ccpo — CHANGES_REQUESTED** [review]

File: f600233d1b154d507b782f84604bcb12/update/sys_script_include_da81226197072110cf3cfd9fe153afee.xml (hunk 1 above)
Comment (jasonburkert-ccpo) [reply f600233d1b154d507b782f84604bcb12/update/sys_script_include_da81226197072110cf3cfd9fe153afee.xml]: good idea

Comment (jasonburkert-ccpo) [reply f600233d1b154d507b782f84604bcb12/update/sys_script_include_da81226197072110cf3cfd9fe153afee.xml]: Removed in https://github.com/dod-ccpo/atat-snow/pull/392/commits/3d7f5576f56fb68d072c873b92739a0f778e2a51

File: f600233d1b154d507b782f84604bcb12/update/sys_script_include_da81226197072110cf3cfd9fe153afee.xml (hunk 3 above)
Comment (jasonburkert-ccpo) [reply f600233d1b154d507b782f84604bcb12/update/sys_script_include_da81226197072110cf3cfd9fe153afee.xml]: I think the additional hints about where things went wrong can be helpful in troubleshooting, but not critical.
Removed try-catches from both helper methods in https://github.com/dod-ccpo/atat-snow/pull/392/commits/3d7f5576f56fb68d072c873b92739a0f778e2a51

Comment (jeffsegal-ccpo) [reply f600233d1b154d507b782f84604bcb12/update/sys_script_include_da81226197072110cf3cfd9fe153afee.xml]: The problem is that you lose the context of the original error, which includes the line number where it occurred. By letting them bubble up, the system logs should have that and actually make it easier to track down. What we see in other files is a `try` followed by 100+ lines of code, so not knowing the specific line makes it far less useful in those cases. Obviously less of an issue here but re-throwing with no additional information is a pretty well known anti-pattern so I'd prefer avoiding it unless there are real benefits.

File: f600233d1b154d507b782f84604bcb12/update/sys_script_include_da81226197072110cf3cfd9fe153afee.xml (hunk 2 above)
Comment (jasonburkert-ccpo) [reply f600233d1b154d507b782f84604bcb12/update/sys_script_include_da81226197072110cf3cfd9fe153afee.xml]: Fixed in https://github.com/dod-ccpo/atat-snow/pull/392/commits/49ed87c5a01b196ffd3b75d2f9c6fc62207c53c8

**Comment by jasonburkert-ccpo** [comment]
@jeffsegal-ccpo I think I've addressed all of your comments.

**Review by jeffsegal-ccpo — APPROVED** [review]

**Pull request merged**